### PR TITLE
feat: get schema by schema_type

### PR DESCRIPTION
**Pull request by Jaycean**
### What this PR does / why we need it:
- part fix apache/apisix#2441

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

File: apisix/admin/plugins.lua [hunk 1]
```diff
@@ -25,7 +25,7 @@ local type      = type
 local table_remove = table.remove
 local table_sort = table.sort
 local table_insert = table.insert
-
+local json_schema = {}
```
Comment (membphis) [inline apisix/admin/plugins.lua:28]: `json_schema` should be a temporary variable in function.

**Review by membphis — CHANGES_REQUESTED** [review]

File: apisix/admin/plugins.lua (hunk 1 above)
Comment (Jaycean) [reply apisix/admin/plugins.lua]: Yes, the previous wording is indeed wrong, it has been revised

**Comment by Jaycean** [comment]
@membphis PTAL,Thx

**Comment by Jaycean** [comment]
@membphis I have some questions I want to discuss with you.

The method I currently implement is to directly determine whether to return `consumer_schema` through whether the query field has `schema_type=consumer` in `apisix/admin/plugins.lua`. 
```
    local arg = get_uri_args()
    local json_schema = plugin.schema	    local json_schema = plugin.schema
    if arg and arg["schema_type"] == "consumer" then
        json_schema = plugin.consumer_schema
    end
```
https://github.com/apache/apisix/issues/2441#issuecomment-714506729
I thought of another implementation method to obtain the corresponding schema directly through the value of `schema_type`, but if Obtain directly through the value of `schema_type`, then you need to pass the corresponding complete schema_type:`/apisix/admin/schema/plugins/{plugin_name}?schema_type=consumer_schema`
Instead of: `/apisix/admin/schema/plugins/{plugin_name}?schema_type=consumer`

**Comment by membphis** [comment]
> /apisix/admin/schema/plugins/{plugin_name}?schema_type=consumer

I think this way is simpler and enough.

**Review by membphis — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]
LGTM

**Comment by liuxiran** [comment]
LGTM

tips:
after this pr merged, jwt-auth also needs to add `consumer_schema` to the `_M` @Jaycean 
https://github.com/apache/apisix/blob/1dfc4cdacd620e6d59c3d4c75506ff80581612eb/apisix/plugins/jwt-auth.lua#L59

**Review by liuxiran — APPROVED** [review]

**Comment by Jaycean** [comment]
> LGTM
> 
> tips:
> after this pr merged, jwt-auth also needs to add `consumer_schema` to the `_M` @Jaycean
> https://github.com/apache/apisix/blob/1dfc4cdacd620e6d59c3d4c75506ff80581612eb/apisix/plugins/jwt-auth.lua#L59

OK, I will submit a new PR unified addition to consumer_schema support

**Comment by juzhiyuan** [comment]
> LGTM
> 
> tips:
> after this pr merged, jwt-auth also needs to add `consumer_schema` to the `_M` @Jaycean
> https://github.com/apache/apisix/blob/1dfc4cdacd620e6d59c3d4c75506ff80581612eb/apisix/plugins/jwt-auth.lua#L59

- [ ] TODO this issue

**Pull request merged**